docs/core/idioms/resource-manager.md

```diff
@@ -0,0 +1,185 @@
+```python
+from prefect import Flow, task, resource_manager, Parameter
+import dask
+from dask.distributed import Client
+
+@resource_manager
+class DaskCluster:
+    """Create a temporary dask cluster.
+
+    Args:
+        - n_workers (int, optional): The number of workers to start.
+    """
+    def __init__(self, n_workers=None):
+        self.n_workers = n_workers
+
+    def setup(self):
+        """Create a temporary dask cluster, returning the `Client`"""
+        return Client(n_workers=self.n_workers)
+
+    def cleanup(self, client):
+        """Shutdown the temporary dask cluster"""
+        client.close()
+
+@task
+def load_data():
+    """Load some data"""
+    return dask.datasets.timeseries()
+
+@task
+def summarize(df, client):
+    """Compute a summary on the data"""
+    return df.describe().compute()
+
+@task
+def write_csv(df, path):
+    """Write the summary to disk as a csv"""
+    return df.to_csv(path, index_label="index")
+
+
+with Flow("dask-example") as flow:
+    n_workers = Parameter("n_workers", default=None)
+    out_path = Parameter("out_path", default="summary.csv")
+
+    with DaskCluster(n_workers=n_workers) as client:
+        # These tasks rely on a dask cluster to run, so we create them inside
+        # the `DaskCluster` resource manager
+        df = load_data()
+        summary = summarize(df, client)
+
+    # This task doesn't rely on the dask cluster to run, so it doesn't need to
+    # be under the `DaskCluster` context
+    write_csv(summary, out_path)
+```
+:::
+
+::: tab "Imperative API"
+```python
+from prefect import Flow, Task, resource_manager, Parameter
+import dask
+from dask.distributed import Client
+
+@resource_manager
+class DaskCluster:
+    """Create a temporary dask cluster.
+
+    Args:
+        - n_workers (int, optional): The number of workers to start.
+    """
+    def __init__(self, n_workers=None):
+        self.n_workers = n_workers
+
+    def setup(self):
+        """Create a temporary dask cluster, returning the `Client`"""
+        return Client(n_workers=self.n_workers)
+
+    def cleanup(self, client):
+        """Shutdown the temporary dask cluster"""
+        client.close()
+
+class LoadData(Task):
+    """Load some data"""
+    def run(self):
+        return dask.datasets.timeseries()
+
+class Summarize(Task):
+    """Compute a summary on the data"""
+    def run(self, df, client):
+        return df.describe().compute()
+
+class WriteCSV(Task):
+    """Write the summary to disk as a csv"""
+    def run(self, df, path):
+        return df.to_csv(path, index_label="index")
+
+
+flow = Flow("dask-example")
+n_workers = Parameter("n_workers", default=None)
+out_path = Parameter("out_path", default="summary.csv")
+
+with DaskCluster(n_workers=n_workers, flow=flow) as client:
+    # These tasks rely on a dask cluster to run, so we create them inside
+    # the `DaskCluster` resource manager
+    df = LoadData()
+    summary = Summarize().bind(df, client, flow=flow)
```

Comment:
Interesting, for some reason I've never considered `bind` a part of the imperative API but it makes sense!